es.go

```diff
@@ -901,7 +911,64 @@ func (c *Client) GetShards(nodes []string) ([]Shard, error) {
 
 	if err != nil {
 		fmt.Printf("Error parsing JSON. Expected ']'")
+		return nil, err
 	}
 
 	return response, nil
 }
+
+//Get details regarding shard distribution across a given set of cluster nodes.
+//
+//Use case: You can leverage this information to determine if it's safe to remove cluster nodes without losing data.
+func (c *Client) GetShardOverlap(nodes []string) (map[string]ShardOverlap, error) {
+	shards, err := c.GetShards(nodes)
+	overlap := map[string]ShardOverlap{}
+
+	if err != nil {
+		fmt.Printf("Error getting shards: %s", err)
+		return nil, err
+	}
+
+	_indices, err := c.GetIndices()
+
+	if err != nil {
+		fmt.Printf("Error getting indices: %s", err)
+		return nil, err
+	}
+
+	// Map-ify this slice of indices for easy lookup
+	indices := map[string]Index{}
+	for _, index := range _indices {
+		indices[index.Name] = index
+	}
+
+	for _, shard := range shards {
+		// Map key is the concatenation of the index name and shard number
+		name := shard.Index + shard.Shard
```

Comment:
Let's change the concatenation here, there is some (low) chance you can hit overlaps. For example, `myindex` has 12 shards and `myindex1` has 2 shards:

* 12th shard of `myindex` would transform to: `myindex12`
* 2nd shard of `myindex1` would transform to `myindex12`

Reply:
Yes, good point.  I've updated to create they map key based on `Index` + "_" + `Shard`